src/rust/inetstack/protocols/tcp/passive_open.rs

```diff
@@ -330,7 +330,12 @@ impl<const N: usize> SharedPassiveSocket<N> {
                 tx_checksum_offload: self.tcp_config.get_rx_checksum_offload(),
             };
             self.transport.transmit(Box::new(segment));
-            self.clock.wait(self.clock.clone(), handshake_timeout).await;
+            let clock_ref: SharedTimer = self.clock.clone();
+            let yielder: Yielder = Yielder::new();
+            if let Err(e) = clock_ref.wait(handshake_timeout, yielder).await {
+                self.ready.push_err(e);
+                return;
+            }
```

Comment:
This code is replicated everwhere. Should we make it a utility standalone function?

Reply:
Yes, I'm going to move the clock to the DemiRuntime and I will move this code as well. The yielder will need to be stored with the queue so we can cancel the coroutine more gracefully if the socket closes and there's cleanup work in the background thread.